service/worker/workerdeployment/version_workflow.go

```diff
@@ -245,28 +250,37 @@ func (d *VersionWorkflowRunner) handleDeleteVersion(ctx workflow.Context) error
 	}
 
 	state := d.GetVersionState()
+	activityCtx := workflow.WithActivityOptions(ctx, defaultActivityOptions)
 
-	// describe all task queues in the deployment, if any have pollers, then cannot delete
-	var tqs []*taskqueuepb.TaskQueue
-	for tqName, _ := range state.TaskQueueFamilies {
-		tqs = append(tqs, &taskqueuepb.TaskQueue{
-			Name: tqName,
-			Kind: enumspb.TASK_QUEUE_KIND_NORMAL, // TODO (Carly): could this be sticky?
-		})
+	// Manual deletion of versions is only possible when:
+	// 1. The version is not current or ramping
+	// 2. The version is drained. (check skipped when `skip-drainage=true` )
+	// 3. The version has no active pollers.
+
+	// 1. Check if the version is not current or ramping.
```

Comment:
technically if a version is drained it should never be current or ramping, but doesn't hurt to check